test/mocks/http/stream_decoder.cc

```diff
@@ -23,6 +23,10 @@ MockRequestDecoder::MockRequestDecoder() {
 MockRequestDecoder::~MockRequestDecoder() = default;
 
 MockResponseDecoder::MockResponseDecoder() {
+  ON_CALL(*this, decode100ContinueHeaders_(_))
+      .WillByDefault(
+          Invoke([](ResponseHeaderMapPtr& headers) { auto moved_headers = std::move(headers); }));
```

Comment:
I'm going to be completely honest in saying that I do not understand why this is required in this change versus my previous one. Without this change, we are hitting asserts in the HTTP/1 codec due to creating the header map twice. AFAICT, this is due to the change in the variant holding the interface pointer vs. the implementation pointer. I think this has to be a compiler bug but I'm not sure.

Reply:
As discussed online will update and add a comment in a follow-up PR